src/cisco_gnmi/nx.py

```diff
@@ -56,12 +56,146 @@ class NXClient(Client):
     >>> capabilities = client.capabilities()
     >>> print(capabilities)
     """
+    def delete_xpaths(self, xpaths, prefix=None):
+        """A convenience wrapper for set() which constructs Paths from supplied xpaths
+        to be passed to set() as the delete parameter.
+
+        Parameters
+        ----------
+        xpaths : iterable of str
+            XPaths to specify to be deleted.
+            If prefix is specified these strings are assumed to be the suffixes.
+        prefix : str
+            The XPath prefix to apply to all XPaths for deletion.
+
+        Returns
+        -------
+        set()
+        """
+        if isinstance(xpaths, string_types):
+            xpaths = [xpaths]
+        paths = []
+        # prefix is not supported on NX yet
+        prefix = None
+        for xpath in xpaths:
+            if prefix:
+                if prefix.endswith("/") and xpath.startswith("/"):
+                    xpath = "{prefix}{xpath}".format(
+                        prefix=prefix[:-1], xpath=xpath[1:]
+                    )
+                elif prefix.endswith("/") or xpath.startswith("/"):
+                    xpath = "{prefix}{xpath}".format(prefix=prefix, xpath=xpath)
+                else:
+                    xpath = "{prefix}/{xpath}".format(prefix=prefix, xpath=xpath)
+            paths.append(self.parse_xpath_to_gnmi_path(xpath))
+        return self.set(deletes=paths)
+
+    def set_json(self, update_json_configs=None, replace_json_configs=None, ietf=False, prefix=None):
```

Comment:
Given NX-OS does not support `prefix` or `ietf` options yet - should we remove? Or will this break current usage of the library?

Reply:
The nexus team implied that they will eventually add support for ietf and prefix so I left them in.  The reality, these APIs are so similar between nx.py and xe.py that I would rather move them all to client.py and just use *args and **kwargs.  I have not had time to implement/propose that and so I pushed this pull request in to get nexus team working on MPTE (which uses your lib).  If we do move to that implementation, the APIs will work when they do add the support.

Reply:
Indeed - the reason these are different now is because of the OS inconsistencies. Once they're all brought in line we can consolidate to `Client`.